src/Raven.Server/Config/Categories/ConfigurationCategory.cs

```diff
@@ -52,34 +52,46 @@ string GetValue(IConfiguration cfg, string name)
                     : null;
             }
 
-            string GetConfiguration(IConfiguration cfg, string name)
+            string GetConfigurationValue(IConfiguration cfg, string keyName, out bool keyExistsInConfiguration)
             {
-                if (cfg == null || name == null)
+                keyExistsInConfiguration = false;
+
+                if (cfg == null || keyName == null)
                     return null;
 
-                var val = GetValue(cfg, name);
+                // This check is needed because cfg.GetSection(keyName) returns null even if key does Not exist in configuration!
+                var keyNames = cfg.AsEnumerable().ToDictionary(x => x.Key, x => x.Value);
```

Comment:
Wouldn't it be better to just do:
```
cfg.AsEnumerable().Any(x => string.Equals(x.Key, keyName))
```

You are already enumerating the cfg anyway.

Also do we need to be case-insensitive during the comparison?

Reply:
Done
The key names within the configuration are Not lower case
I don't think it must be a case-insensitive check